lib/coxir/struct/channel.ex

```diff
@@ -160,6 +162,146 @@ defmodule Coxir.Struct.Channel do
     API.request(:post, "channels/#{channel}/messages/bulk-delete", %{messages: messages})
   end
 
+  @doc """
+  Modifies the name of a given channel.
+
+  Returns a channel object upon success
+  or a map containing error information.
+  """
+  @spec set_name(channel, String.t) :: map
+
+  def set_name(%{id: id}, name),
+    do: set_name(id, name)
+
+  def set_name(channel, name) do
+    edit(channel, %{name: name})
+  end
+
+  @doc """
+  Modifies the topic of a given channel.
+
+  Returns a channel object upon success
+  or a map containing error information.
+  """
+  @spec set_topic(channel, String.t) :: map
+
+  def set_topic(%{id: id}, topic),
+    do: set_topic(id, topic)
+
+  def set_topic(channel, topic) do
+    edit(channel, %{topic: topic})
+  end
+
+  @doc """
+  Enables the NSFW status of a given channel.
+
+  Returns a channel object upon success
+  or a map containing error information.
+  """
+  @spec enable_nsfw(channel) :: map
+
+  def enable_nsfw(%{id: id}),
```

Comment:
Wouldn't a `set_nsfw/2` make more sense?
Instead of having both an enable and a disable function.

Reply:
I could probably agree taking into account that we are using setters' style for function naming, but maybe for user end it may be more useful to know you can dis/en-able some properties? (this comment also applies to set_hoist and set_mentionable comments